lib/driver.rb

```diff
@@ -0,0 +1,82 @@
+require_relative 'csv_record'
+
+module RideShare
+  class Driver < CsvRecord
+    attr_reader :name, :vin, :trips, :status
+
+    def initialize(id:, name:, vin:, trips: [], status: :AVAILABLE)
+      super(id)
+
+      @name = name
+      @vin = vin
+      @trips = trips
+      @status = status.to_sym
+
+      check_status(status)
+      check_vin(vin)
+    end
+
+    def check_vin(vin)
+      if vin.length < 17 || vin.length > 17
+        raise ArgumentError.new("Invalid vin")
+      end
+    end
+
+    def check_status(status)
+      valid_status = [:AVAILABLE, :UNAVAILABLE]
+      if valid_status.include?(status.to_sym) == false
+        raise ArgumentError.new("Invalid status")
+      else
+        return true
+      end
+    end
```

Comment:
Love these validation methods! Thorough validation without cluttering up the constructor :)